### PR TITLE
Add error handling and specify encodings.

### LINKED ISSUE #91: Handling malformed UTF8
As of now, everything assumes all words are proper utf8. Perhaps add a lossy arg analogous to `finalfusion-rust` in the read method(s).

### PR DESCRIPTION
**Pull request by sebpuetz**
There's some noise from refactoring the scripts, mostly due the odd design of  `Format.{to, from}`.

### CODE REVIEW

File: src/finalfusion/scripts/util.py [hunk 1]
```diff
@@ -32,6 +30,24 @@ def add_format_args(parser: ArgumentParser, short: str, name: str,
...
 
 
+def add_common_args(parser: ArgumentParser):
+    parser.add_argument(
+        "-l",
+        "--lossy",
+        action="store_true",
+        default=False,
+        help=
+        "Whether to fail on malformed UTF8. Setting this flag replaces malformed UTF8"
```
Comment (danieldk) [inline src/finalfusion/scripts/util.py]: UTF-8

File: src/finalfusion/compat/text.py [hunk 2]
```diff
@@ -29,18 +30,24 @@ def load_text_dims(file: Union[str, bytes, int, PathLike]) -> Embeddings:
     Parameters
     ----------
     file : str, bytes, int, PathLike
-        Path to a file with embeddings in word2vec binary format.
+        Path to a file with embeddings in text format with dimensions on the first line.
+    lossy : bool
+        If set to true, malformed UTF8 sequences in words will be replaced with the `U+FFFD`
```
Comment (danieldk) [inline src/finalfusion/compat/text.py]: UTF-8

File: src/finalfusion/compat/text.py [hunk 3]
```diff
@@ -57,14 +64,18 @@ def load_text(file: Union[str, bytes, int, PathLike]) -> Embeddings:
     ----------
     file : str, bytes, int, PathLike
         Path to a file with embeddings in word2vec binary format.
+    lossy : bool
+        If set to true, malformed UTF8 sequences in words will be replaced with the `U+FFFD`
```
Comment (danieldk) [inline src/finalfusion/compat/text.py]: UTF-8

File: src/finalfusion/compat/word2vec.py [hunk 4]
```diff
@@ -30,6 +31,9 @@ def load_word2vec(file: Union[str, bytes, int, PathLike]) -> Embeddings:
     ----------
     file : str, bytes, int, PathLike
         Path to a file with embeddings in word2vec binary format.
+    lossy : bool
+        If set to true, malformed UTF8 sequences in words will be replaced with the `U+FFFD`
```
Comment (danieldk) [inline src/finalfusion/compat/word2vec.py]: UTF-8

**Review by danieldk — APPROVED** [review]
Looks good, but maybe you could replace *UTF8* in the documentation by *UTF-8*, which is the name of the standard ;).

**Comment by sebpuetz** [comment]
Thanks, fixed!

**Pull request merged**